### PR TITLE
Intital pass at a printf implementation

### LINKED ISSUE #41: Implement printf

### PR DESCRIPTION
**Pull request by CaseyB**
This PR is a first pass at implementing printf to resolve #41 
#HappyHacktoberfest !

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ members = [
     "nice",
     "nl",
     "nohup",
+    "printf",
```
Comment (GrayJack) [inline Cargo.toml:34]: Can you add this to all <Platform>.toml files? Just so the CI can better test your code

Comment (CaseyB) [reply Cargo.toml:34]: Done. I also marked printf as in progress in the README

Comment (GrayJack) [reply Cargo.toml:34]: Great! Have fun implementing it!

**Comment by CaseyB** [comment]
Sorry about that, the author info got screwed up so I had to update the commits.

**Comment by CaseyB** [comment]
Is there anything I'm missing? I believe this covers the core printf functionality.

File: printf/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
+mod cli;
+
+use std::{process, result::Result, str::FromStr};
+
+use clap::Values;
+
+const ARG_UNDERFLOW_ERROR: i32 = 64;
+const ARG_UNPARSABLE_ERROR: i32 = 64;
```
Comment (GrayJack) [inline printf/src/main.rs]: Since those are different errors, can they have different values?

File: README.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ cargo install --path .
 |    od    |      X      |         |      |
 |  paste   |      X      |         |      |
 |  patch   |      X      |         |      |
-|  printf  |      X      |         |      |
+|  printf  |             |    X    |      |
```
Comment (GrayJack) [inline README.md]: Since it is already POSIX compatible, you can go ahead and say it is done

**Review by GrayJack — COMMENTED** [review]
Thanks for contributing!! 🎉 

The code looks solid, I just had one nitpick related to the exit code and a commentary on the table

**Comment by GrayJack** [comment]
Awesome! Thanks for the contribution 🎉 

bors r+

**Pull request merged**